modules/vault-cluster/main.tf

```diff
@@ -108,35 +111,37 @@ resource "aws_autoscaling_group" "autoscaling_group" {
 # CREATE LAUNCH CONFIGURATION TO DEFINE WHAT RUNS ON EACH INSTANCE IN THE ASG
 # ---------------------------------------------------------------------------------------------------------------------
 
-resource "aws_launch_configuration" "launch_configuration" {
+resource "aws_launch_template" "launch_template" {
   name_prefix   = "${var.cluster_name}-"
   image_id      = var.ami_id
   instance_type = var.instance_type
   user_data     = var.user_data
 
   iam_instance_profile = aws_iam_instance_profile.instance_profile.name
   key_name             = var.ssh_key_name
-  # TF-UPGRADE-TODO: In Terraform v0.10 and earlier, it was sometimes necessary to
-  # force an interpolation expression to be interpreted as a list by wrapping it
-  # in an extra set of list brackets. That form was supported for compatibilty in
-  # v0.11, but is no longer supported in Terraform v0.12.
-  #
-  # If the expression in the following list itself returns a list, remove the
-  # brackets to avoid interpretation as a list of lists. If the expression
-  # returns a single list item then leave it as-is and remove this TODO comment.
   security_groups = concat(
     [aws_security_group.lc_security_group.id],
     var.additional_security_group_ids,
   )
-  placement_tenancy           = var.tenancy
-  associate_public_ip_address = var.associate_public_ip_address
+
+  placement {
+    tenancy = var.tenancy
+  }
+
+  network_interfaces {
+    associate_public_ip_address = var.associate_public_ip_address
+  }
 
   ebs_optimized = var.root_volume_ebs_optimized
 
-  root_block_device {
-    volume_type           = var.root_volume_type
-    volume_size           = var.root_volume_size
-    delete_on_termination = var.root_volume_delete_on_termination
+  block_device_mappings {
+    device_name = "/dev/xvda"
+    ebs {
+      delete_on_termination = var.root_volume_delete_on_termination
+      encrypted             = true
```

Comment:
The launch config & launch template are essentially identical except for `encrypted = true`.

Reply:
Do we have to indicate that this is the root volume, or is that assumed now?